### PR TITLE
Cleanup CSS variables

### PR DESCRIPTION
**Pull request by youknowriad**
Some cleaning to our CSS variables.

 - Rename icon-button-size to just button-size
 - Remove deprecated grid units
 - A few other tweaks

### CODE REVIEW

File: packages/block-editor/src/components/colors-gradients/style.scss [hunk 1]
```diff
@@ -1,11 +1,11 @@
 .block-editor-color-gradient-control {
 	&__color-indicator {
-		margin-bottom: $grid-size;
+		margin-bottom: $grid-unit;
```
Comment (jasmussen) [inline packages/block-editor/src/components/colors-gradients/style.scss]: I don't think we should ever actually use $grid-unit, it should always be one of the numbered ones. I.e. in this case, it should be $grid-unit-10.

In that vein, $grid-unit is intended only to seed all the numbered variables. Should it have a different name to reflect that?

File: packages/block-editor/src/components/default-block-appender/style.scss [hunk 2]
```diff
@@ -48,7 +48,6 @@
 
 // Ensure that the height of the first appender, and the one between blocks, is the same as text.
 .block-editor-default-block-appender__content {
-	min-height: $empty-paragraph-height / 2;
```
Comment (jasmussen) [inline packages/block-editor/src/components/default-block-appender/style.scss]: I haven't had time to test this yet, are you sure there aren't any negataive consequences of removing this? I'm sure we added it for a reason, back in the day.

**Comment by jasmussen** [comment]
In general this is a really nice bit of cleanup, thank you. I promise I'll get to the color variable cleanup as well, perhaps in my color PR. I left two comments to address.

File: packages/block-editor/src/components/default-block-appender/style.scss (hunk 2 above)
Comment (youknowriad) [reply packages/block-editor/src/components/default-block-appender/style.scss]: The reason we added it is to make sure the appender with or without paragraph has the same height. It's broken with or without this style now.

File: packages/block-editor/src/components/colors-gradients/style.scss (hunk 1 above)
Comment (youknowriad) [reply packages/block-editor/src/components/colors-gradients/style.scss]: I updated but why not just remove the variable in that case and use $grid-unit-10 everywhere.

File: packages/block-editor/src/components/default-block-appender/style.scss (hunk 2 above)
Comment (jasmussen) [reply packages/block-editor/src/components/default-block-appender/style.scss]: Not sure I follow but I'll check when I have a moment.

File: packages/block-editor/src/components/colors-gradients/style.scss (hunk 1 above)
Comment (jasmussen) [reply packages/block-editor/src/components/colors-gradients/style.scss]: Oh I guess that could work? Seems almost cyclical 😅

The alternative would be to spell out 8px for each variable.

**Pull request merged**